src/input_xml.F90

```diff
@@ -2169,15 +2170,15 @@ subroutine read_tallies_xml()
 
       ! Copy tally id
       if (check_for_node(node_tal, "id")) then
-        call get_node_value(node_tal, "id", t % id)
+        call get_node_value(node_tal, "id", tally_id)
```

Comment:
If you move the call to `openmc_tally_set_id` immediately after this, then `t % id` can be used as is below.